Mac-App/Backend/Backend/BackendAPI/Models/DependencyNode.swift

```diff
@@ -36,3 +36,17 @@ public final class DependencyNode: Equatable {
         dependency.parent = self
     }
 }
+
+// MARK: - Debug description
+
+extension DependencyNode: CustomStringConvertible {
+    public var description: String {
+        var text = "\(name)"
+
+        if !dependencies.isEmpty {
```

Comment:
You don't do `== false` anymore after that tweet? :D @Thurman1776-

Reply:
Haha, I try to remember whenever possible

Reply:
@emrepun 
I guess it depends on the context too. 
I realised I had introduced an endless recursive expression when implementing `Equatable` for a reference type. Looking at the stack trace, I found that it was crashing [here](https://github.com/Thurman1776-/Uncomplect-us/blob/master/Mac-App/Backend/Backend/Parsers/SwiftDepsParser/SwiftDepsParser.swift#L33) where I am still doing `== false`. In this case, if I even if I change it to `!` it still goes through the operator overwrite so the compiler still needs to do some work I think. 

(I fixed it by comparing the parent's name instead of the whole reference)
https://github.com/Thurman1776-/Uncomplect-us/pull/36/commits/bd046a6840524aece40e85ae1ec98bef8a32872c

Reply:
Interesting, since the crash is related to an endless recursion, do you think it matters where exactly it crashes? Also nice fix! @Thurman1776-

Reply:
@emrepun 
For this case I think it does as the equality operator overload was calling itself infinitely. From what I understand here, it did not make a difference if I would use `result.contains(tree) == false` or `!result.contains(tree)` as the overload is called anyways. 

So it seems that for custom equatable overloads, the swift compiler has to do some additional work anyways (to find the proper implementation). So perhaps for built-in types using `!` instead of `== false` does really help.

Reply:
I see it makes sense, thanks for the explanation @Thurman1776-